contracts/Minter.sol

```diff
@@ -0,0 +1,116 @@
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/utils/SafeERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/cryptography/MerkleProofUpgradeable.sol";
+
+interface Daomon721a {
+    function mint(address to, uint256 amount) external;
+}
+
+contract Minter is OwnableUpgradeable {
+    using SafeERC20Upgradeable for IERC20Upgradeable;
+
+    error NoPurchasableTokens();
+    error ZeroAmountNotAllowed();
+    error NotEnoughMintableTokensForThisUser();
+    error ReceiverNotWhitelisted();
+
+    // User variables
+    uint256 public numOfPurchasedTokens;
+    mapping(address => uint256) public numOfMintableTokensPerAddress;
+
+    // Admin variables
+    uint256 public deploymentTimeStamp;
+    uint256 public priceInPMON;
+    uint256 public dailySupply;
+    address public pmonAddress;
+    address public pmonReceiverAddress;
+    address public daomon721aAddress;
+    bytes32 public merkleRoot;
+
+    function initialize(
+        address _pmonReceiverAddress,
+        address _daomon721aAddress,
+        bytes32 _merkleRoot
+    ) public initializer {
+        deploymentTimeStamp = block.timestamp;
+        dailySupply = 10;
+        priceInPMON = 10e18;
+        pmonAddress = 0x1796ae0b0fa4862485106a0de9b654eFE301D0b2;
+        pmonReceiverAddress = _pmonReceiverAddress;
+        daomon721aAddress = _daomon721aAddress;
+        merkleRoot = _merkleRoot;
+
+        __Ownable_init();
+    }
+
+    function purchaseTo(
+        address to,
+        uint256 amount,
+        bytes32[] calldata proof
+    ) external {
+        /* Checks */
+        if (getNumOfPurchasableTokens() - amount <= 0)
+            revert NoPurchasableTokens();
+        if (amount <= 0) revert ZeroAmountNotAllowed();
+
+        bytes32 leaf = keccak256(abi.encodePacked(to));
+        if (!MerkleProofUpgradeable.verifyCalldata(proof, merkleRoot, leaf))
+            revert ReceiverNotWhitelisted();
+
+        /* Effects */
+        numOfPurchasedTokens += amount;
+        numOfMintableTokensPerAddress[to] += amount;
```

Comment:
Do we want to add a buy or mint limit per user per day? Otherwise one whitelisted person could buy all tokens of a day? 

If this is automated someone might get a huge amount of shares in the beginning, nor?

Reply:
Yeah this is no problem imo